### PR TITLE
Remove 3.0 test docs from 4.0

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Remove v3.0-related test docs from `release-4.0`. The aliases are added in https://github.com/pingcap/docs/pull/3182.

### Which TiDB version(s) do your changes apply to? (Required)

- [ ] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3904
- Other reference link(s):

### CODE REVIEW

File: faq/tidb-faq.md [hunk 1]
```diff
@@ -695,7 +695,7 @@ WAL belongs to ordered writing, and currently, we do not apply a unique configur
 
 #### How is the write performance in the most strict data available mode (`sync-log = true`)?
 
-Generally, enabling `sync-log` reduces about 30% of the performance. For write performance when `sync-log` is set to `false`, see [Performance test result for TiDB using Sysbench](/benchmark/v3.0-performance-benchmarking-with-sysbench.md).
+Generally, enabling `sync-log` reduces about 30% of the performance. For write performance when `sync-log` is set to `false`, see [Performance test result for TiDB using Sysbench](https://docs.pingcap.com/tidb/v3.0/v3.0-performance-benchmarking-with-sysbench).
```
Comment (lilin90) [inline faq/tidb-faq.md:698]: - `sync-log` is not mentioned in https://docs.pingcap.com/tidb/v3.0/v3.0-performance-benchmarking-with-sysbench. It seems that the content needs update. Please help confirm it.
- Consider using the corresponding file for v4.0: https://docs.pingcap.com/tidb/v4.0/benchmark-sysbench-v4-vs-v3

File: faq/tidb-faq.md [hunk 2]
```diff
@@ -759,7 +759,7 @@ It is recommended to use TiDB server if you query a single table mainly using fi
 At the beginning, many users tend to do a benchmark test or a comparison test between TiDB and MySQL. We have also done a similar official test and find the test result is consistent at large, although the test data has some bias. Because the architecture of TiDB differs greatly from MySQL, it is hard to find a benchmark point. The suggestions are as follows:
 
 - Do not spend too much time on the benchmark test. Pay more attention to the difference of scenarios using TiDB.
-- See [Performance test result for TiDB using Sysbench](/benchmark/v3.0-performance-benchmarking-with-sysbench.md).
+- See [Performance test result for TiDB using Sysbench](https://docs.pingcap.com/tidb/v3.0/v3.0-performance-benchmarking-with-sysbench).
```
Comment (lilin90) [inline faq/tidb-faq.md:762]:
```suggestion
- See [Performance test result for TiDB using Sysbench](/benchmark/benchmark-sysbench-v4-vs-v3).
```

File: faq/tidb-faq.md (hunk 1 above)
Comment (TomShawn) [reply faq/tidb-faq.md:698]: I find out that `syn-log` is mentioned only in [TiDB Sysbench 性能对比测试报告 - v2.1 对比 v2.0](https://docs.pingcap.com/zh/tidb/v2.1/v2.1-performance-benchmarking-with-sysbench) (zh and en), while in other versions, `syn-log` is not mentioned in sysbench test report. I think `see Performance test result for TiDB using Sysbench` can be deleted in `dev`, `v4.0`, `v3.1`, and `v3.0`. I'll create PRs to do that.

File: faq/tidb-faq.md [hunk 3]
```diff
@@ -759,7 +759,7 @@ It is recommended to use TiDB server if you query a single table mainly using fi
 At the beginning, many users tend to do a benchmark test or a comparison test between TiDB and MySQL. We have also done a similar official test and find the test result is consistent at large, although the test data has some bias. Because the architecture of TiDB differs greatly from MySQL, it is hard to find a benchmark point. The suggestions are as follows:
 
 - Do not spend too much time on the benchmark test. Pay more attention to the difference of scenarios using TiDB.
-- See [Performance test result for TiDB using Sysbench](/benchmark/v3.0-performance-benchmarking-with-sysbench.md).
+- See [Performance test result for TiDB using Sysbench](/benchmark/benchmark-sysbench-v4-vs-v3).
```
Comment (TomShawn) [inline faq/tidb-faq.md:762]:
```suggestion
- See [Performance test result for TiDB using Sysbench](/benchmark/benchmark-sysbench-v4-vs-v3.md).
```

**Comment by TomShawn** [comment]
@lilin90 PTAL again, thanks!

**Pull request merged**